projects/core/directives/dropdown/dropdown-context.directive.ts

```diff
@@ -44,6 +62,48 @@ export class TuiDropdownContextDirective extends TuiRectAccessor {
         this.driver.next(false);
     }
 
+    @HostListener('touchmove.silent', [
+        '$event.touches[0].clientX',
+        '$event.touches[0].clientY',
+    ])
+    onTouchMove(x: number, y: number): void {
+        if (!this.isIOS) {
+            return;
+        }
+
+        this.currentRect = tuiPointToClientRect(x, y);
+    }
+
+    @HostListener('touchstart.silent', [
+        '$event.touches[0].clientX',
+        '$event.touches[0].clientY',
+    ])
+    onTouchStart(x: number, y: number): void {
+        if (!this.isIOS) {
+            return;
+        }
+
+        this.currentRect = tuiPointToClientRect(x, y);
+        this.zone.runOutsideAngular(() => {
+            this.longTapTimeout = this.win.setTimeout(() => {
+                this.zone.run(() => {
+                    this.driver.next(true);
+                });
+            }, tuiLongTapDelay);
+        });
+    }
+
+    @HostListener('touchend.silent')
+    @HostListener('touchcancel.silent')
+    onTouchEnd(): void {
+        if (!this.isIOS) {
+            return;
+        }
+
+        clearTimeout(this.longTapTimeout);
+        this.driver.next(false);
```

Comment:
Wouldn't that close the dropdown when we release the finger?

Reply:
It looks like a bug in TuiPortalService.remove.
Because DropdownContextDirective emits false, but dropwown still present until next interaction.

<img width="633" alt="image" src="https://github.com/taiga-family/taiga-ui/assets/143048525/362e4443-b691-4b47-8251-97d4a3f9eb43">

It fixes the issue.

Reply:
fixed